tests/test_circulator_reports.py

```diff
@@ -1,179 +1,237 @@
 
-import pandas as pd
+import pytest
+import pandas as pd  # type: ignore
 from sqlalchemy.orm import Session  # type: ignore
 
 from transitstat.circulator.schema import CirculatorArrival, CirculatorBusRuntimes, CirculatorRidership
 from transitstat.circulator.reports import parse_args, RidesystemReports
-from .factories import ArrivalFactory
 
 
+@pytest.mark.filterwarnings("ignore::sqlalchemy.exc.SAWarning")
+@patch('transitstat.circulator.reports.RideSystemsInterface')
+def test_get_otp_no_force(mocked_rs_cls, conn_str, arrival_factory, arrival_dataset):
+    """Test for get_otp without force"""
 
-def test_factory(fake_session):
-    """Test for factory"""
-    ArrivalFactory()
-    ArrivalFactory(scheduled_arrival_time=time(12, 0))
+    # arrival_factory.create_batch(100)
+    test_df = pd.DataFrame(data=arrival_dataset.create_batch(10))
 
-    assert fake_session.query(CirculatorArrival).count() == 2
+    inst = RidesystemReports(conn_str, 'username', 'superdupersecretpassword')
+    inst.rs_cls.get_otp.return_value = test_df
 
-@patch('transitstat.circulator.reports.RideSystemsInterface')
-def test_get_otp_no_force(mocked_rs_cls, conn_str):
-    """Test for get_otp"""
-    test_df = {
-        'date': [date(2022, 1, 1), date(2022, 1, 1), date(2022, 1, 1), date(2022, 1, 1)],
-        'route': ['xx', 'xx', 'xx', 'xx'],
-        'stop': ['xx', 'xx', 'xx', 'xx'],
-        'blockid': ['xx', 'xx', 'xx', 'xx'],
-        'actualarrivaltime': [time(), time(), time(), time()],
-        'actualdeparturetime': [time(), time(), time(), time()],
-        'scheduledarrivaltime': [time(), time(1), time(2), time(3)],
-        'scheduleddeparturetime': [time(), time(), time(), time()],
-        'vehicle': ['xx', 'xx', 'xx', 'xx'],
-        'ontimestatus': ['xx', 'xx', 'xx', 'xx']
-        }
-
-    inst = RidesystemReports(conn_str)
-    inst.rs_cls._login.return_value = "Login successful :)"
-    inst.rs_cls.get_otp.return_value = pd.DataFrame(test_df)
+    with Session(bind=inst.engine, future=True) as session:
+        batch = arrival_factory.create_batch(100)
+        session.bulk_save_objects(batch)
+        session.commit()
 
     inst.get_otp(date(2022, 1, 1), date(2022, 1, 1), hours='12')
+    assert mocked_rs_cls.assert_called_once_with('username', 'superdupersecretpassword') is None
+    assert inst.rs_cls.get_otp.assert_called_once_with(date(2022, 1, 1), date(2022, 1, 1), hours='12') is None  # pylint: disable=no-member
+
+    with Session(bind=inst.engine, future=True) as session:
+        ret = session.query(CirculatorArrival)
+        assert ret.count() == 110
+
+    test_df['stop'] = '25th St'
+    inst.get_otp(date(2022, 1, 1), date(2022, 1, 1))
     with Session(bind=inst.engine, future=True) as session:
         ret = session.query(CirculatorArrival)
-        assert ret.count() == 8
+        assert ret.count() == 110
 
+    assert ['25th St' for _ in range(10)] != [r.stop for r in session.query(CirculatorRidership).all()[100:110]]
```

Comment:
First, you can do `['25th St'] * 10` to make a list of the same thing. But I am not a big fan of asserting that it isn't equal, because this would pass if nothing was returned as well, or if they were different lengths. I am also not completely sure that you should rely on the return being the same order every time.